### PR TITLE
Throwing fix

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

Fixes https://github.com/ParadiseSS13/Paradise/issues/24084
In no gravity movement is never blocked to allow you use jetpack and stopping by objects nearby
In gravity movement is still blocked(if thats intended by throw type) so nothing will stop the throw

## Why It's Good For The Game

Bugs bad, using jetpack in space to stop movement is good. Using magboots to stop the throw is good.

## Testing

Compiled, tried some space throwing

## Changelog
:cl:
fix: You can stop in no gravity area with jetpack and magboots once again.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/SSthrowing.dm [hunk 1]
```diff
@@ -72,6 +72,8 @@ SUBSYSTEM_DEF(throwing)
 	var/dodgeable = TRUE
 	/// Can a thrown mob move themselves to stop the throw?
 	var/block_movement = TRUE
+	/// This is internal variable to support no gravity movement even if variable above is TRUE. This variable will change during throwing
+	var/block_movement_actual = TRUE
```
Comment (lewcc) [inline code/controllers/subsystem/SSthrowing.dm:76]: please use a better variable name here

Comment (HMBGERDO) [reply code/controllers/subsystem/SSthrowing.dm]: *deathgasp

File: code/controllers/subsystem/SSthrowing.dm [hunk 2]
```diff
@@ -71,6 +71,8 @@ SUBSYSTEM_DEF(throwing)
 	///When this variable is false, non dense mobs will be hit by a thrown item. useful for things that you dont want to be cheesed by crawling, EG. gravitational anomalies
 	var/dodgeable = TRUE
 	/// Can a thrown mob move themselves to stop the throw?
+	var/should_block_movement = TRUE
+	/// This is internal variable to support no gravity movement even if variable above is TRUE. This variable will change during throwing
```
Comment (S34NW) [inline code/controllers/subsystem/SSthrowing.dm:75]: `The variable above` isn't helpful for popups when the var is accessed elsewhere, please tweak it

Comment (HMBGERDO) [inline code/controllers/subsystem/SSthrowing.dm:75]:
```suggestion
	/// This is internal variable to support no gravity movement even if var/should_block_movement is TRUE. This variable will change during throwing
```

File: code/controllers/subsystem/SSthrowing.dm [hunk 3]
```diff
@@ -71,6 +71,8 @@ SUBSYSTEM_DEF(throwing)
 	///When this variable is false, non dense mobs will be hit by a thrown item. useful for things that you dont want to be cheesed by crawling, EG. gravitational anomalies
 	var/dodgeable = TRUE
 	/// Can a thrown mob move themselves to stop the throw?
+	var/should_block_movement = TRUE
+	/// This is internal variable to support no gravity movement even if var/should_block_movement is TRUE. This variable will change during throwing
 	var/block_movement = TRUE
```
Comment (Contrabang) [inline code/controllers/subsystem/SSthrowing.dm:76]: If this is an internal-only variable, we should have spacemanDMM enforce it
```suggestion
	VAR_PRIVATE/block_movement = TRUE
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**